src/hooks/useWorkletStateMachine.ts

```diff
@@ -0,0 +1,166 @@
+import {useCallback} from 'react';
+import {runOnJS, runOnUI, useSharedValue} from 'react-native-reanimated';
+import Log from '@libs/Log';
+
+// When you need to debug state machine change this to true
+const DEBUG_MODE = true;
```

Comment:
You shouldn't really need this flag I don't think. Just use `Log.client` (I don't think we want these logs sent to server).

We currently have Logs enabled in production, but I don't think there's really a performance concern with logs that's in-scope for this issue.

Reply:
Again this code from @terrysahaidak 

@terrysahaidak what do you think?

Reply:
Having these logs is only helpful when you debug these state machines, in other cases there will be just a bunch of noise in the logs every time you interact with the keyboard and other things that interact with the state machine.

Another reason for having this flag was to avoid unnecessary calls from the UI thread where we run all the worklets logic to the JS thread where we log stuff, which can affect performance.

Reply:
I think we should change it to `Log.client` but keep that flag (but set it to `false`), because it would really bloat the logs if we set it to true. And it's good to keep it in case you ever need to debug it.

Although we could consider just covering it with a unit test, and then as this is quite isolated code it shouldn't really cause any bugs.